contrib/babelfishpg_tsql/src/pltsql_utils.c

```diff
@@ -1574,6 +1576,71 @@ split_object_name(char *name)
 	return res;
 }
 
+/*
+ * Wrapper over split_object_name function above to expose it as a SQL function.
+ */
+Datum
+split_identifier_internal(PG_FUNCTION_ARGS)
+{
+	FuncCallContext	*funcctx;
+	char        	**split_parts = NULL;
+
+	if (SRF_IS_FIRSTCALL())
+	{
+		int         	num_parts = 0;
+		MemoryContext	oldcontext;
+
+		funcctx = SRF_FIRSTCALL_INIT();
+		oldcontext = MemoryContextSwitchTo(funcctx->multi_call_memory_ctx);
+		if (!PG_ARGISNULL(0))
+		{
+			char	*input;
+			char	**splited_object_name;
+			int 	i, j = 0;
+
+			input = text_to_cstring(PG_GETARG_TEXT_P(0));
+			splited_object_name = split_object_name(input);
+
+			for (i = 0; i < 4; i++)
+			{
+				if (strlen(splited_object_name[i]) > 0)
+					num_parts++;
+			}
+
+			if (num_parts > 0)
+			{
+				split_parts = (char **) palloc(num_parts * sizeof(char *));
```

Comment:
Is there any specific reason that this memory should be alloc in multi_call_memory_ctx ?

We should either make sure this array would be useful throughout multi calls or we should remember free it before this function returns.